students/AndrewMiotke/session09/mailroom_oo/mailroom.py

```diff
@@ -0,0 +1,153 @@
+#!/usr/bin/env python3
+import os
+import sys
+import time
+import shutil
+from textwrap import dedent
+from donors_model import Donor, DonorList
+
+get_donor_list = DonorList()
+
+
+def report():
+    print("📊 Generating report...")
+    time.sleep(1)
+    print_donor_report()
+
+
+def print_donor_report():
+    report_rows = []
+
+    for (donor_name, gifts) in get_donor_list.donor_data.values():
+        total_gifts = sum(gifts)
+        num_gifts = len(gifts)
+        avg_gift = total_gifts / num_gifts
+        report_rows.append((donor_name, total_gifts, num_gifts, avg_gift))
+
+    report_rows.sort(key=sort_key, reverse=True)
+
+    print("{:25s} | {:11s} | {:9s} | {:12s}".format("Donor Name", "Total Given", "Num Gifts", "Average Gift"))
+    print("-" * 66)
+
+    for row in report_rows:
+        print("{:25s}   {:11.2f}   {:9d}   {:12.2f}".format(*row))
+
+
+def print_donors():
```

Comment:
This is something that could be implemented within the DonorList class itself. Since it has access to all its donors, it could handle the printing instead of putting the logic in your mailroom. 

In general, the logic for the donors should go into that file while logic for the donor_list should go into its class.

Good use of the models but this is something to keep in mind for future OOP! The goal is to put the logic/functions of a specific class into its own class. 

Nice work overall!